### DISCUSSION TITLE
Pyinstaller artifact built with Github Action becomes Document kind in MacOS

### DISCUSSION THREAD

**Discussion by Kalsyc** [pyinstaller/pyinstaller#7131] [Help] [answered]
Hi, I have been scratching my head about this and I'm not sure if it's because of some code signing issues. I'm trying to build the pyinstaller app for windows/macos/linux for my work. 

Whilst manually building it via terminal is fine for all 3 operating systems... the problem comes when I use a github action to automate this and have it output as an artifact for all 3 environments. The ubuntu-latest and windows-latest environment works fine, and I managed to get the executable out and working. However for MacOS, the executable becomes a 'Document' kind when downloaded instead of 'Unix Executable File'

<img width="623" alt="image" src="https://user-images.githubusercontent.com/45655404/194049897-de5ec892-2a98-442b-af60-b17f2716ce07.png">

`vti_gen` is when I build the app locally
`vti_gen_t` is when i download it as an artifact (which was built on a github action macos-latest env)

I checked both files using the `file -I` command and sure enough...
<img width="382" alt="image" src="https://user-images.githubusercontent.com/45655404/194050128-91fd6f51-6951-4769-a3c6-6df61557004a.png">
<img width="361" alt="image" src="https://user-images.githubusercontent.com/45655404/194050148-cc5bf7c7-e6d2-4a05-95bc-0038e9a8188b.png">

Is this a problem with the code signing or permissions? I am unable to run the file built in the github action for macos. Thanks for the help!

**Comment by rokm** [comment, marked as answer]
Depending on how you downloaded the file, I suppose
a) it could have lost its executable permissions. `chmod +x vti_gen_t` from terminal should fix that.
b) it could have been flagged with `com.apple.quarantine` due to being downloaded. Check by running `xattr vti_gen_t`, and clear with `xattr -d com.apple.quarantine vti_gen_t`, if necessary.

**Reply by Kalsyc** [reply]
You are right. chmod +x works. Thanks! I just thought it was strange how macOS views the file as a 'Document' when there's no access... didn't know that it just be default behaviour.
